pkg/event/event.go

```diff
@@ -0,0 +1,62 @@
+package event
```

Comment:
This should be internal/event, exposed via apidef.TykEvent. Err on the side of less API changes / no new API additions, and composing the data model with type aliases against the internal packages. This lets us add utilities/tests that focus particularly on events, eventually moving the dispatcher logic